util/dao/utils/src/lib.rs

```diff
@@ -25,36 +25,57 @@ pub enum Error {
     Format,
 }
 
-pub fn genesis_dao_data(genesis_cellbase_tx: &TransactionView) -> Result<Byte32, FailureError> {
-    let c = genesis_cellbase_tx
-        .data()
-        .raw()
-        .outputs()
-        .into_iter()
-        .try_fold(Capacity::zero(), |capacity, output| {
-            let cap: Capacity = output.capacity().unpack();
-            capacity.safe_add(cap)
-        })?;
-    let u = genesis_cellbase_tx
-        .data()
-        .raw()
-        .outputs()
+pub fn genesis_dao_data(
+    txs: Vec<&TransactionView>,
+    // All input occupied capacities in txs
+    inputs: Vec<Capacity>,
```

Comment:
It is simple to figure out which outputs have been spent in the genesis block. We just need to index all the inputs in the transaction view first and use that to filter out the outputs, like this:

```
let dead_cells = HashSet::new();
for tx in txs {
  for i in tx.inputs {
      dead_cells.insert(i.out_point)
  }
}
for tx in txs {
  for (i, o) in tx.outputs.enumerate() {
    if !dead_cells.contains((tx, i)) {
      count_c_and_u(o);
    }
  }
}
```